app/helpers/parse_dd20.py

```diff
@@ -0,0 +1,860 @@
+        station_acline_names = data_station.acline_name_list
+        line_acline_names = data_line.acline_name_list
+
+        # Verfify that the same names are present in both station and line data, since else DD20 format has an error.
+        names_in_station_but_not_line = list(
+            set(station_acline_names).difference(line_acline_names)
+        )
+        if names_in_station_but_not_line:
+            raise ValueError(
+                f"The following AC-line names are present in station "
+                + "but not line sheet of DD20: {names_in_station_but_not_line}."
+            )
+
+        names_in_line_but_not_station = list(
+            set(line_acline_names).difference(station_acline_names)
+        )
+        if names_in_line_but_not_station:
+            raise ValueError(
+                f"The following AC-line names are present in line "
+                + "but not station sheet of DD20: {names_in_line_but_not_station}."
+            )
+
+        # Init AC-Line properties mapping dictionarys via methods on objects
+        acline_name_to_conductor_count = data_station.get_conductor_count_dict()
+        acline_name_to_system_count = data_station.get_system_count_dict()
+        acline_name_to_conductor_type = data_station.get_conductor_type_dict()
+        acline_name_to_conductor_max_temp = data_station.get_conductor_max_temp_dict()
+        acline_name_to_cablelim_continuous = data_station.get_cablelim_continuous_dict()
+        acline_name_to_cablelim_15m = data_station.get_cablelim_15m_dict()
+        acline_name_to_cablelim_1h = data_station.get_cablelim_1h_dict()
+        acline_name_to_cablelim_40h = data_station.get_cablelim_40h_dict()
+        acline_name_to_lim_continuous = data_line.get_acline_lim_continuous_dict()
+        acline_name_to_complim_continuous = data_line.get_complim_continuous_dict()
+        acline_name_to_complim_15m = data_line.get_complim_15m_dict()
+        acline_name_to_complim_1h = data_line.get_complim_1h_dict()
+        acline_name_to_complim_40h = data_line.get_complim_40h_dict()
+
+        # Init kv mapping dict (only use from station since we just checked on agrement on line names)
+        st_acline_name_to_conductor_kv_level = (
+            data_station.get_conductor_kv_level_dict()
+        )
+
+        """
+        Translation of AC-line names from DD20 datasource to desired format using Regex.
+        I.e 132 kV line with data source name "EEE-FFF-2" is translated to "E_EEE-FFF_2"
+
+        Regex expression does the following:
+        (?P<STN1>\\w{3,4}?) makes a group 'STN1' and input a word between 3-4 chars
+        (?P<STN2>\\w{3,4}?) makes a group 'STN2' and input a word between 3-4 chars
+        (?P<id>\\w)? makes a group 'id' and if there is a word in the end of the name it stores it
+        """
+        acline_name_to_translated_name = {}
+        none_translated_acline_name = []
+        REGEX = r"^(?P<STN1>\w{3,4})-(?P<STN2>\w{3,4})-?(?P<id>\w)?$"
```

Comment:
```suggestion
        REGEX = r"^(?P<STN1>\w{3,4})-(?P<STN2>\w{3,4})(-(?P<id>\w))?$"
```
I would argue this is more "correct". "Yours" will accept ABC-DEF34 - "mine" will require the second hyphen.

Be aware that \w will accept numbers as well - is this good, or bad?
we COULD replace "\w" with "[A-Za-zØÆÅøæå]" if we want to make sure it will only be standard alphabetical chars (including Danish custom chars).

```suggestion
        REGEX = r"^(?P<STN1>\w{3,4})-(?P<STN2>\w{3,4})(-(?P<id>\d))?$"
```

If the last group (id) should only accept numerical values, you can use \d in stead.